src/main/java/com/google/javascript/clutz/DeclarationGenerator.java

```diff
@@ -246,6 +251,19 @@ public String produceDts(Depgraph depgraph) {
     return out.toString();
   }
 
+  private Set<String> getShadowedProvides(Set<String> provides) {
+    Set<String> shadowedProvides = new TreeSet<>();
```

Comment:
This looks a bit as if a `Trie` was the better data structure, but I guess it doesn't matter for O(100k) symbols?